### PR TITLE
Add support for server sent events via response asStream

### PR DESCRIPTION
**Pull request by adamsmo**
Before submitting pull request:
- [x] Check if the project compiles by running `sbt compile`
- [x] Verify docs compilation by running `sbt compileDocs`
- [x] Check if tests pass by running `sbt test`
- [x] Format code by running `sbt scalafmt`

docs are still WIP, but the implementation is finished

### CODE REVIEW

File: akka-http-backend/src/main/scala/sttp/client3/akkahttp/AkkaHttpServerSentEvents.scala [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import akka.stream.scaladsl.{Flow, Framing}
+import akka.util.ByteString
+import sttp.client3.sse.ServerSentEvent
+
+object AkkaHttpServerSentEvents {
+
+  def decodeSSE: Flow[ByteString, ServerSentEvent, NotUsed] =
+    //todo decide on max frame length
+    Framing
+      .delimiter(ByteString("\n\n"), maximumFrameLength = Int.MaxValue, allowTruncation = true)
```
Comment (adamsmo) [inline akka-http-backend/src/main/scala/sttp/client3/akkahttp/AkkaHttpServerSentEvents.scala:13]: `maximumFrameLength` can be parameter accepted by `decodeSSE` method

Comment (adamw) [reply akka-http-backend/src/main/scala/sttp/client3/akkahttp/AkkaHttpServerSentEvents.scala:13]: Is there anything specified by the SSE standard? Maximum message size? If not, we should make this configurable, with some reasonable default - maybe taken from other impls?

File: httpclient-backend/zio/src/main/scala/sttp/client3/httpclient/zio/ZioServerSentEvents.scala [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package sttp.client3.httpclient.zio
+
+import sttp.capabilities.zio.ZioStreams
+import sttp.client3.sse.ServerSentEvent
+import zio.stream.ZTransducer
+
+object ZioServerSentEvents {
```
Comment (adamw) [inline httpclient-backend/zio/src/main/scala/sttp/client3/httpclient/zio/ZioServerSentEvents.scala:7]: this should go to `implementations/zio` as well, like with `fs2` (thus avoiding duplication)

File: testing/server/src/main/scala/sttp/client3/testing/server/HttpServer.scala [hunk 3]
```diff
@@ -105,6 +106,29 @@ private class HttpServer(port: Int, info: String => Unit) extends AutoCloseable
           parameterMap { _ => entity(as[String]) { (body: String) => complete(body) } }
         }
       }
+    } ~ pathPrefix("sse") {
+      path("echo") {
```
Comment (adamw) [inline testing/server/src/main/scala/sttp/client3/testing/server/HttpServer.scala:110]: it's not really an `echo` :) but rather `replicate3` or sth like that

Comment (adamw) [reply testing/server/src/main/scala/sttp/client3/testing/server/HttpServer.scala:110]: or an `echo3`

File: core/src/main/scala/sttp/client3/sse/ServerSentEvent.scala [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+case class ServerSentEvent(
+    data: Option[String] = None,
+    eventType: Option[String] = None,
+    id: Option[String] = None,
+    retry: Option[Int] = None
+)
+
+object ServerSentEvent {
+  def parseEvent(event: List[String]): ServerSentEvent = {
```
Comment (adamw) [inline core/src/main/scala/sttp/client3/sse/ServerSentEvent.scala:13]: maybe add some tests?
it's not possible for parsing to fail - there are no invalid inputs?

**Comment by adamw** [comment]
@adamsmo Are you planning to continue working on this PR, or should I take over? :)

**Comment by adamsmo** [comment]
Please take over, responding to your last comment

>maybe add some tests?
it's not possible for parsing to fail - there are no invalid inputs?

It should not be possible for parsing to fail, as it ignores any value with not recognized label. If the line does not contain any `:` it is treated as a line with label and no value, and if the label is unknown, it is ignored.

I think I do not have push permission to this branch after I was removed from the SML organization

**Comment by adamw** [comment]
Released in [3.0.0-RC13](https://github.com/softwaremill/sttp/releases/tag/v3.0.0-RC13)

**Pull request merged**